src/Microsoft.AspNet.OData.Shared/Routing/Conventions/NestedPathsRoutingConvention.cs

```diff
@@ -0,0 +1,89 @@
+using Microsoft.AspNetCore.Mvc.Controllers;
+using Microsoft.AspNetCore.Mvc.Infrastructure;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.OData.Edm;
+using Microsoft.OData.UriParser;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Microsoft.AspNet.OData.Routing.Conventions
+{
+    /// <summary>
+    /// An implementation of <see cref="IODataRoutingConvention"/> that handles GET requests
+    /// to arbitrarily nested paths. It requires a controller action with the [EnableNestedPaths] attribute.
+    /// This should be the last convention in order not to override other routing conventions that
+    /// rely on user-defined methods.
+    /// </summary>
+    public partial class NestedPathsRoutingConvention : NavigationSourceRoutingConvention
+    {
+        /// <summary>
+        /// Selects the action for OData requests.
+        /// </summary>
+        /// <param name="odataPath">The OData path.</param>
+        /// <param name="controllerContext">The controller context.</param>
+        /// <param name="actionMap">The action map.</param>
+        /// <returns>
+        ///   <c>null</c> if the request isn't handled by this convention; otherwise, the name of the selected action
+        /// </returns>
+        internal static string SelectActionImpl(ODataPath odataPath, IWebApiControllerContext controllerContext, IWebApiActionMap actionMap)
+        {
+
+            ODataRequestMethod method = controllerContext.Request.GetRequestMethodOrPreflightMethod();
+
+            if (method != ODataRequestMethod.Get)
+            {
+                // [EnableNestedPaths] only supports GET requests
+                return null;
+            }
+
+
+            // unsupported path segments
+            if (odataPath.PathTemplate.EndsWith("$ref")
+                || odataPath.PathTemplate.EndsWith("$value")) // TODO: check whether we support $value
+            {
+                return null;
+            }
+
+            ODataPathSegment firstSegment = odataPath.Segments.FirstOrDefault();
+
+            string sourceName;
+            if (firstSegment is EntitySetSegment entitySetSegment)
+            {
+                sourceName = entitySetSegment.EntitySet.Name;
+            }
+            else if (firstSegment is SingletonSegment singletonSegment)
+            {
+                sourceName = singletonSegment.Singleton.Name;
+            }
+            else
+            {
+                // this only supports paths starting with an entity set or singleton
+                return null;
+            }
+
+
+
+
+            // if we did no find a matching action amongst the conventional user-defined methods
+            // then let's check if the controller has a Get method with [EnableNestedPaths] attribute
+            // which should be used to catch any any nested GET request
+            IWebApiActionDescriptor descriptor = actionMap.GetActionDescriptor($"Get{sourceName}") ?? actionMap.GetActionDescriptor("Get");
```

Comment:
I'm not sure I fully understand the question, but assuming we have both a `Get()` and a `Get(int key)` method. Then the `Get(int key)` will be handled by the `EntityRoutingConvention`. Because this convention runs after the other conventions have been tried, the overloads should have been taken care by this point.